aptos-move/framework/supra-framework/sources/vesting_without_staking.move

```diff
@@ -390,60 +390,69 @@ module supra_framework::vesting_without_staking {
-                emit_event(
-                    &mut vesting_contract.vest_events,
-                    VestEvent {
-                        admin: vesting_contract.admin,
-                        shareholder_address: shareholder_address,
-                        vesting_contract_address: contract_address,
-                        period_vested: next_period_to_vest,
-                    },
-                );
-                //update last_vested_period for the shareholder
-                vesting_record.last_vested_period = next_period_to_vest;
-                next_period_to_vest = next_period_to_vest + 1;
+        let vesting_contract = borrow_global<VestingContract>(contract_address);
+        // Short-circuit if vesting hasn't started yet.
+        if (vesting_contract.vesting_schedule.start_timestamp_secs > timestamp::now_seconds()) {
+            return
         };
 
+        let vesting_record = simple_map::borrow(&vesting_contract.shareholders,&shareholder_address);
+
+        // Check if the next vested period has already passed. If not, short-circuit since there's nothing to vest.
+        let vesting_schedule = vesting_contract.vesting_schedule;
+        let last_vested_period = vesting_record.last_vested_period;
+        let next_period_to_vest = last_vested_period + 1;
+        let last_completed_period =
+            (timestamp::now_seconds() - vesting_schedule.start_timestamp_secs) / vesting_schedule.period_duration;
+        while(last_completed_period>=next_period_to_vest) {
+            vest_transfer(&mut next_period_to_vest, contract_address, shareholder_address);
         };
 	}
 
+    fun vest_transfer(next_period_to_vest: &mut u64, contract_address: address,
+                      shareholder_address: address) acquires VestingContract {
+        let vesting_signer = account::create_signer_with_capability(&borrow_global<VestingContract>(contract_address).signer_cap);
+        //extract beneficiary address
+        let beneficiary = beneficiary(contract_address,shareholder_address);
+
+        let vesting_contract = borrow_global_mut<VestingContract>(contract_address);
+        let vesting_record = simple_map::borrow_mut(&mut vesting_contract.shareholders,&shareholder_address);
+
+        // Check if the next vested period has already passed. If not, short-circuit since there's nothing to vest.
+        let vesting_schedule = vesting_contract.vesting_schedule;
+        let schedule = &vesting_schedule.schedule;
+
+        // Index is 0-based while period is 1-based so we need to subtract 1.
+        let schedule_index = *next_period_to_vest - 1;
+        let vesting_fraction = if (schedule_index < vector::length(schedule)) {
+            *vector::borrow(schedule, schedule_index)
+        } else {
+            // Last vesting schedule fraction will repeat until the grant runs out.
+            *vector::borrow(schedule, vector::length(schedule) - 1)
+        };
+
```

Comment:
I think perhaps this code can either go in a separate function, that takes `next_period_to_vest` and returns `vesting_fraction`, or it can go in the caller `vest_individual` so that this particular function merely does the transfer as per given `vesting_fraction`.